sleuth/sleuth/enrich_tls.py

```diff
@@ -89,9 +102,102 @@ def check_compliance(compliance_policy, scs):
                 return "yes" if scs_desc in compliance_data[compliance_policy] else "no"
 
     return "error loading file"
+    
+    
+def audit_certs_issuer(certs, trusted_ca_list):
+    # as implemented now, we will get the tls certs in order so that the root
+    # cert is at the end. We check to see if the issuer is in our trusted_ca_list;
+    # if not, we report it as a concern
+    for each in certs[-1]['issuer']:
+        if each['entry_id'] == 'organizationName':
+            if each['entry_data'] not in trusted_ca_list:
+                return 'CA not trusted: ' + each['entry_data']
+    
+    return None
+
+
+def get_seclevel_with_param(policy, attr, value, param):
+    current_seclevel = None
+
+    if param:
+        if attr == "kex":
+            param_rules = policy.rules[attr][value]["client_key_length"]
+        elif attr == "sig_alg":
```

Comment:
Changing this to "cert_sig_alg" upstream would help to communicate to the reader exactly what it is.